src/data/bmg_plate.py

```diff
@@ -0,0 +1,109 @@
+import os
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+
+
+class Plate:
+    """
+    Class representing a plate with values resulting from an HTS experiment
+    """
+    def __init__(self, filepath: str) -> None:
+        """
+        :param filepath: path to the file consisting plate
+        """
+        self.filepath = filepath
+        self.barcode = filepath.split('/')[-1].split('.')[0].split('\\')[-1]
+        self.plate_array = self.parse_file()
+        self.control_statistics()
+        self.z_factor = 1 - (3*(self.std_pos+self.std_neg) /
+                             (self.mean_neg-self.mean_pos))
+
+    def parse_file(self) -> np.array:
+        """
+        Read data from txt file to np.array
+
+        :return: array with plate values
+        """
+        plate = np.zeros(shape=(16, 24))
+        with open(self.filepath) as f:
+            for i, line in enumerate(f.readlines()):
+                # handle different formatting (as in FIREFLY)
+                if i == 0 and 'A' not in line:
+                    df = pd.read_csv(self.filepath, header=None)
+                    plate = df.to_numpy()
+                    break
+                well, value = line.split()
+                i, j = self.well_to_ids(well)
+                plate[i, j] = value
+        return plate
+
+    def well_to_ids(self, well_name: str) -> tuple[int, int]:
+        """
+        Helper method to map well name to index
+
+        :param well_name: well name in format {letter}{number} (e.g. A10)
+        to be transformed
+        """
+        head = well_name.rstrip('0123456789')
+        tail = well_name[len(head):]
+        return ord(head)-65, int(tail)-1
+
+    def control_statistics(self) -> None:
+        """
+        Calculate statistic for control values e.g. two last columns of a plate
+        """
+        pos = self.plate_array[:, -2]
+        neg = self.plate_array[:, -1]
+        self.std_pos = np.std(pos)
+        self.std_neg = np.std(neg)
+        self.mean_pos = np.mean(pos)
+        self.mean_neg = np.mean(neg)
+
+    def find_outliers(self):
+        # TODO
+        # append to df then, add visualization
+        ...
+
+    def summary_row(self) -> list:
+        """
```

Comment:
suggestion: for summary a dictionary with keys describing the values would be easier to work with later on (we would not have to remember which value corresponds to which stat)
or even better - a `NamedTuple`/`dataclass` called PlateSummary

Reply:
yeah that's right but we need that method only to add row to df, where columns are already defined

do you think it is still necessary to name the stats in that case?